src/Sarif/IAnalysisContext.cs

```diff
@@ -29,5 +29,11 @@ public interface IAnalysisContext : IDisposable
         bool AnalysisComplete { get; set; }
 
         DefaultTraces Traces { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum file size (in kilobytes) that will be analyzed.
+        /// If not set, it will analyze all sizes.
+        /// </summary>
+        int FileSizeInKilobytes { get; set; }
```

Comment:
>FileSizeInKilobytes [](http://example.com/codeflow?start=12&length=19)

since we are here, can we update this to MaxFileSizeInKilobytes or something that actually explains why is this needed?

Reply:
This was mentioned in the other PR, it would be a breaking change for SPAM. Is that acceptable?

Reply:
I think, if that would make things clearer and we tell people, yes.

What are your thoughts?

Reply:
Another strategy could be the following:
1. we rename the C# property
2. we keep the exposed argument.

So, internally, we know the good name. and we tell that in the next release we will rename the argument :)

Reply:
Please do not take this change now. A breaking change of this kind needs to come with a thoughtful implementation, this is new work that is slowing down the immediate change.

So, go ahead and file an issue. We have other patterns of deprecating command-line args (like --hashes is support to be obsoleted!!). We do this in a manner such as Eddy describes, retain the old and mark it obsolete and/or update the code to properly transform old command-lines to new. 

So, Ed, be sure to capture your excellent thoughts in the issue that you file. :)